mlflow/java/src/main/java/org/mlflow/sagemaker/ScoringServer.java

```diff
@@ -0,0 +1,270 @@
+package org.mlflow.sagemaker;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.mlflow.mleap.MLeapLoader;
+import org.mlflow.models.Model;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+/** A RESTful webserver for {@link Predictor Predictors} that runs on the local host */
+public class ScoringServer {
+  public static final String RESPONSE_KEY_ERROR_MESSAGE = "Error";
+
+  public static final int HTTP_RESPONSE_CODE_SERVER_ERROR = 500;
+  public static final int HTTP_RESPONSE_CODE_SUCCESS = 200;
+  public static final int DEFAULT_PORT = 5001;
+
+  private enum RequestContentType {
+    Csv("text/csv"),
+    Json("application/json"),
+    Invalid("invalid");
+
+    private final String value;
+    private static final Map<String, RequestContentType> BY_VALUE_MAP = new HashMap<>();
+
+    static {
+      for (RequestContentType inputType : RequestContentType.values()) {
+        BY_VALUE_MAP.put(inputType.value, inputType);
+      }
+    }
+
+    RequestContentType(String value) {
+      this.value = value;
+    }
+
+    public static RequestContentType fromValue(String value) {
+      if (BY_VALUE_MAP.containsKey(value)) {
+        return BY_VALUE_MAP.get(value);
+      } else {
+        return RequestContentType.Invalid;
+      }
+    }
+  }
+
+  private final Optional<Predictor> predictor;
+  private final Optional<Integer> portNumber;
+  private Optional<Service> activeService = Optional.empty();
```

Comment:
I would recommend marking this `@Nullable` and dealing with that, since `activeService=None` is an entirely internal state (it just simplifies the syntax without increasing the risk significantly), but this is just an opinion, so feel free to ignore.

Reply:
Removed. We now use `org.eclipse.jetty.server.Server.isStarted()` to determine activity.